generate/spec.go

```diff
@@ -1,37 +1,109 @@
 package generate
 
 import (
-	bane "github.com/jessfraz/bane/apparmor"
+    "os"
+    "io"
+    "path"
+    "text/template"
 )
 
 // Declaration holds all the data from karn declaration files
 type Declaration struct {
-	SystemCalls  SystemCalls  `toml:"System-Calls,omitempty"`
-	Capabilities Capabilities `toml:"Capabilities,omitempty"`
-	Filesystem   FileSystem   `toml:"Filesystem,omitempty"`
-	Network      Network      `toml:"Network,omitempty"`
-	System       System       `toml:"System,omitempty"`
+    SystemCalls  SystemCalls  `toml:"System-Calls,omitempty"`
+    Capabilities CapConfig    `toml:"Capabilities,omitempty"`
+    Filesystem   FsConfig     `toml:"Filesystem,omitempty"`
+    Network      NetConfig    `toml:"Network,omitempty"`
+    System       System       `toml:"System,omitempty"`
 }
 
 // System holds OS/Arch specific arguments
 type System struct {
-	Architectures        []string `toml:"Architectures,omitempty"`
-	DefaultSyscallAction string   `toml:"DefaultSyscallAction,omitempty"`
+    Architectures        []string `toml:"Architectures,omitempty"`
+    DefaultSyscallAction string   `toml:"DefaultSyscallAction,omitempty"`
 }
 
 // Syscalls to handle with each supported action
 type SystemCalls struct {
-	Allow []string `toml:"Allow,omitempty"`
-	Trap  []string `toml:"Trap,omitempty"`
-	Trace []string `toml:"Trace,omitempty"`
-	Kill  []string `toml:"Kill,omitempty"`
-	Errno []string `toml:"Errno,omitempty"`
+    Allow []string `toml:"Allow,omitempty"`
+    Trap  []string `toml:"Trap,omitempty"`
+    Trace []string `toml:"Trace,omitempty"`
+    Kill  []string `toml:"Kill,omitempty"`
+    Errno []string `toml:"Errno,omitempty"`
 }
 
+/**
+ * Originally Written by Jess Frazelle
+ * http://github.com/jessfraz/bane
+ */
+type AppArmorProfileConfig struct {
+    Name         string
+    Filesystem   FsConfig
+    Network      NetConfig
+    Capabilities CapConfig
 
-// Type aliases for use from bane, temporary 
+    Imports      []string
```

Comment:
We're actually not using the Imports or InnerImports fields, you can remove them.